hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/services/AbfsOutputStream.java

```diff
@@ -389,6 +423,12 @@ private synchronized void flushWrittenBytesToServiceAsync() throws IOException {
 
   private synchronized void flushWrittenBytesToServiceInternal(final long offset,
       final boolean retainUncommitedData, final boolean isClose) throws IOException {
+
+    // flush is not called for appendblob as is not needed
+    if (this.isAppendBlob) {
+      return;
```

Comment:
nit: Should we have a debug log here?

Reply:
this can lead to frequent log lines, every time a flush(), hflush(), hsync() is called. will that be ok?

Reply:
Resolved